articles/implementing-caching-in-nodejs-using-redis/index.md

```diff
@@ -0,0 +1,331 @@
+
+To install the necessary dependencies, in your cloned folder execute the following command:
+
+```bash
+cd ./start && npm install
+```
+
+Our main focus shall be implementing `fetchFoodItem` function in `controllers/recipe.js` file. It handles the logic for the `recipe/:foodItem` route.
+
+### Fetching data from recipe labs API.
+
+To fetch data from the API we:
+
+- Import `axios` module:
+
+```javascript
+const axios = require("axios");
+```
+
+- Modify the `fetchFoodItem` function as follows:
+
+```javascript
+
+const fetchFoodItem = (req,res,next) => {
+
+    try {
+
+        //destructure the foodItem from params.
+
+        let { foodItem } = req.params;
+
+        //fetch the data.
+
+        const recipe = await axios
+        .get(`http://www.recipepuppy.com/api/?q=${foodItem}`)
+        .catch(console.log);
+
+        //return a response.
+        return res.send({
+          success: true,
+          message: recipe.data.results,
+        });
+
+      } catch (err) {
+
+        //return the error
+        return res.send({
+          success: false,
+          message: err,
+        });
+
+      };
+
+};
+
+```
+
+From above:
+
+- Get the food item from the request object.
+- Fetch data from the public API.
+- Send the data to the client.
+
+To test this, follow the following steps:
+
+- Start the development server from your terminal by running: `npm run dev`.
+
+- Head over to Postman and send a request to the route. Do not include the semi-colons. The request should be similar to: `http://localhost:3000/recipe/coffee`. Feel free to replace coffee with your preference.
+
+- Wait for the response to be sent back and then check the time taken on the upper right section. Typically, it will take up to 600 ms or even more. This amount of time means that our application is consuming a lot of user network bandwidth. To minimize this cost, we have to set up `caching`. For this article, we will use `redis.`
+
+### Installing redis.
+
+If you have already installed `redis`, feel free to proceed to the next step. Else follow along the below steps.
+
+- Open a separate tab in your terminal and run the following command to shift to your home directory.
+
+```bash
+cd
+```
+
+- In your home directory, run the following commands one by one:
+
+```bash
+wget http://download.redis.io/redis-stable.tar.gz
+tar xvzf redis-stable.tar.gz
+cd redis-stable
+make
+sudo make install
+```
+
+- To confirm installation, run the following command to start the `redis server`:
+
+```bash
+redis-server
+```
+
+- If the server is successfully started, continue to the next step. Else if you encounter an error, kindly reference from the [official docs](https://redis.io/topics/quickstart).
+
+### Importing and configuring redis.
+
+Follow the below steps:
+
+- Import `redis` module.
```

Comment:
It's important to specify to which file I'm importing Redis.
At the moment, the best I could do is make an educated guess, but that might not be the case for a reader trying to work with Redis for the first time.